core/src/test/scala/org/apache/spark/sql/delta/DeltaSuite.scala

```diff
@@ -873,113 +904,106 @@ class DeltaSuite extends QueryTest
-            .save(tempDir.getCanonicalPath)
+        Seq(1, 2, 3).toDF
+          .withColumn("part", $"value" % 2)
+          .write
+          .format("delta")
+          .partitionBy("part")
+          .mode("append")
+          .save(tempDir.getCanonicalPath)
 
-          Seq(1, 5).toDF
-            .withColumn("part", $"value" % 2)
-            .write
-            .format("delta")
-            .partitionBy("part")
-            .mode("overwrite")
-            .save(tempDir.getCanonicalPath)
-          checkDatasetUnorderly(data.select("value").as[Int], 1, 2, 5)
-        }
+        Seq(1, 5).toDF
+          .withColumn("part", $"value" % 2)
+          .write
+          .format("delta")
+          .partitionBy("part")
+          .option(DeltaOptions.PARTITION_OVERWRITE_MODE_OPTION, "static")
+          .mode("overwrite")
+          .save(tempDir.getCanonicalPath)
+        checkDatasetUnorderly(data.select("value").as[Int], 1, 5)
       }
     }
   }
 
-  test(
-    "batch write: append, dynamic partition overwrite set via conf and overridden via option") {
+  test("batch write: append, overwrite without partitions should ignore partition overwrite mode") {
     withSQLConf(DeltaSQLConf.DYNAMIC_PARTITION_OVERWRITE_ENABLED.key -> "true") {
-      withSQLConf(SQLConf.PARTITION_OVERWRITE_MODE.key -> "dynamic") {
-        withTempDir { tempDir =>
-          def data: DataFrame = spark.read.format("delta").load(tempDir.toString)
+      withTempDir { tempDir =>
+        def data: DataFrame = spark.read.format("delta").load(tempDir.toString)
 
-          Seq(1, 2, 3).toDF
-            .withColumn("part", $"value" % 2)
-            .write
-            .format("delta")
-            .partitionBy("part")
-            .mode("append")
-            .save(tempDir.getCanonicalPath)
+        Seq(1, 2, 3).toDF
+          .withColumn("part", $"value" % 2)
+          .write
+          .format("delta")
+          .mode("append")
+          .save(tempDir.getCanonicalPath)
 
-          Seq(1, 5).toDF
-            .withColumn("part", $"value" % 2)
-            .write
-            .format("delta")
-            .partitionBy("part")
-            .option(DeltaOptions.PARTITION_OVERWRITE_MODE_OPTION, "static")
-            .mode("overwrite")
-            .save(tempDir.getCanonicalPath)
-          checkDatasetUnorderly(data.select("value").as[Int], 1, 5)
-        }
+        Seq(1, 5).toDF
+          .withColumn("part", $"value" % 2)
+          .write
+          .format("delta")
+          .mode("overwrite")
+          .option(DeltaOptions.PARTITION_OVERWRITE_MODE_OPTION, "dynamic")
+          .save(tempDir.getCanonicalPath)
+        checkDatasetUnorderly(data.select("value").as[Int], 1, 5)
       }
     }
   }
 
-  test(
-    "batch write: append, overwrite without partitions should ignore partition overwrite mode") {
+  test("batch write: append, overwrite non-partitioned table with replaceWhere ignores partition " +
+    "overwrite mode option") {
```

Comment:
this is a new test. the diffs are messed up again, and `"batch write: append, overwrite without partitions should ignore partition overwrite mode"` is above... all the test(...) { are shifted down one with their contents